src/sentry/issues/escalating_group_forecast.py

```diff
@@ -55,6 +55,13 @@ def fetch(cls, project_id: int, group_id: int) -> Optional[EscalatingGroupForeca
             date_added=datetime.now(),
         )
 
+    @classmethod
+    def fetch_todays_forecast(cls, project_id: int, group_id: int) -> int:
+        date_now = datetime.now().date()
+        escalating_forecast = EscalatingGroupForecast.fetch(project_id, group_id)
+        forecast_today_index = (date_now - escalating_forecast.date_added.date()).days
```

Comment:
Is this not causing a bug in the tests? Am I doing something wrong?

```
> from datetime import datetime
> date_now = datetime.now().date()  # Line 60
> print(date_now)
2023-04-26
> date_now - datetime.now()
TypeError: unsupported operand type(s) for -: 'datetime.date' and 'datetime.datetime'
> type(date_now)
datetime.date
> type(datetime.now())
datetime.datetime
```

Random typing thing I've noticed. Why is `datetime` different between the two classes?
https://github.com/getsentry/sentry/blob/82e7fe0bdfb482b11743c0ec0552d8ac02fbc2ad/src/sentry/issues/escalating_group_forecast.py#L24
https://github.com/getsentry/sentry/blob/82e7fe0bdfb482b11743c0ec0552d8ac02fbc2ad/src/sentry/issues/escalating_group_forecast.py#L37

Could you check if we have typing issues in here?

Here are some notes I was working on that may be useful in the future
----------------------------------------------------------------------

I can't, at first read, if there would be any sort of bug in here. You probably know most of what I will mention here, however, for people looking at this PR later it will be good background knowledge.

In the large majority of the cases, a group that is marked as "archived until escalating" will *NOT* have a group forecast because we will save storing 95% of all groups in the node store.

In such cases, we will fetch the default low ceiling 
https://github.com/getsentry/sentry/blob/82e7fe0bdfb482b11743c0ec0552d8ac02fbc2ad/src/sentry/issues/escalating_group_forecast.py#L51-L56

Reply:
**Addressing `forecast_today_index = (date_now - escalating_forecast.date_added.date()).days`:**
- Both `date_now` and `escalating_forecast.date_added.date()` are of type `datetime.date()` so this issue does not occur (please let me know if this is not what you were referring to in your comment)

**Addressing the differing types for `date_added`**:
- Objects of type `datetime` are not serializable and thus cannot be stored in nodestore. I work around this by converting it to a float timestamp and then back to a datetime. This is how it is done for the `detection_time` field in  [IssueOccurrence](https://github.com/getsentry/sentry/blob/a506567bd35fd0cd4a7678080d3ff8418313b128/src/sentry/issues/issue_occurrence.py)